### PR TITLE
DOC: minor changes to the documentation

### PR DESCRIPTION
**Pull request by tacaswell**
I suspect the last commit will be the only contentious one.

### CODE REVIEW

**Comment by takluyver** [comment]
I think with the release automation, the WIP release notes are meant to be in `news/` - @scopatz, can you confirm exactly what we need there?

**Comment by scopatz** [comment]
Yeah, you should follow the instructions here: https://regro.github.io/rever-docs/news

File: docs/whatsnew/3.0.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+What's new in h5py 3.0
+======================
```
Comment (scopatz) [inline docs/whatsnew/3.0.rst]: rever will create this file at release time, so no need to add it here

Comment (scopatz) [reply docs/whatsnew/3.0.rst]: (ie please remove it)

File: docs/whatsnew/index.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ These document the changes between minor (or major) versions of h5py.
 
 .. toctree::
 
+    3.0
```
Comment (scopatz) [inline docs/whatsnew/index.rst]: Likewise, rever will add this link automatically, so please remove it

**Comment by scopatz** [comment]
That all seems reasonable to me

**Comment by takluyver** [comment]
I've just fixed a typo which would probably have caused an error when the release notes were built into docs.

We already have a CI job that builds docs - could we set that up to pull together release notes using rever first, so any invalid markup in the release notes is flagged automatically?

**Comment by tacaswell** [comment]
@aragilar Thanks for cleaning up after me!  I've been super busy/sick since I got home.

@takluyver I agree, getting CI warnings about invalid rst as close to when it goes in is important.

**Pull request merged**